core/arch/arch.c

```diff
@@ -3550,6 +3550,15 @@ priv_mcontext_to_dr_mcontext(dr_mcontext_t *dst, priv_mcontext_t *src)
      */
     if (dst->size > sizeof(dr_mcontext_t))
         return false;
+#if defined(AARCH64)
+    /* Clients built before support of Arm AArch64's Scalable Vector Extension
+     * (SVE) are not binary compatible with the latest build.
+     */
+    if (TEST(DR_MC_MULTIMEDIA, dst->flags) && dst->size != sizeof(dr_mcontext_t)) {
+        CLIENT_ASSERT(
+            false, "A pre-SVE client is running on an Arm AArch64 SVE DynamoRIO build!");
```

Comment:
Let's also bump VERSION_NUMBER_DEFAULT in the top-level CMakeLists.txt to 9.94 and bump OLDEST_COMPATIBLE_VERSION_DEFAULT in the same file to "994".

Reply:
Updated `VERSION_NUMBER_DEFAULT` and `OLDEST_COMPATIBLE_VERSION_DEFAULT` in top-level `CMakeLists.txt` as well as `VERSION_NUMBER` in `ci-docs.yml` and `ci-package.yml`.

Is the size check ok as it is raising a `CLIENT_ASSERT` or should we `return false`?

Reply:
Seems ok to have a debug build immediate stop.